src/common/shared_config.h

```diff
@@ -11,14 +11,28 @@
 #define FW_UPDATE_ENABLE  0xAA
 #define FW_UPDATE_DISABLE 0x00
 
-// pin PA13 state
-#define APPENDIX_FLAG_MASK 0x01
+typedef enum {
+    BT_APPENDIX_EXISTS = 0, /**< Appendix not broken, signature authentication needed */
+	BT_APPENDIX_BROKEN	/**< Appendix broken, no signature authentication needed */
+} BT_APPENDIX_STATUS;
+
+// model_specific_flags bit locations from LSB
+typedef enum {
+	BT_APPENDIX_STATUS_LOC = 0,
+	BT_BOOT_FLAG_1,
+	BT_BOOT_FLAG_2,
+	BT_BOOT_FLAG_3,
+	BT_BOOT_FLAG_4,
+	BT_BOOT_FLAG_5,
+	BT_BOOT_FLAG_6,
+	BT_BOOT_FLAG_7
+} BT_BOOT_FLAGS_LOC;
 
 #pragma pack(push)
 #pragma pack(1)
 typedef struct {
     uint8_t fw_update_flag;
-    uint8_t model_specific_flags; // ~ "reserved1" originally
+    uint8_t boot_flags;
     uint8_t reserved2;
     uint8_t reserved3;
     uint8_t reserved4;
```

Comment:
/nitpick/
Why not 

```C
typedef enum {
    FW_UPDATE_ENABLE = 0xAA,
    FW_UPDATE_DISABLE = 0xFF,
} fw_update_flag;

typedef struct  {
    fw_update_flag       fw_update_flag;
    bool                 appendix_broken : 1;
    uint8_t              __reserved[14];
} data_exchange_t;
```

Less code, clearer and safer code, the same memory layout.

Reply:
yes this is better implementation, thanks for pointing out